scripts/zephyr_module.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Tool for parsing a list of projects to determine if they are Zephyr
+projects. If no projects are given then the output from `west list` will be
+used as project list.
+
+Include file is generated for Kconfig using --kconfig-out.
+A <name>:<path> text file is generated for use with CMake using --cmake-out.
+'''
+
+import argparse
+import os
+import sys
+import yaml
+import pykwalify.core
+import subprocess
+import re
+
+
+METADATA_SCHEMA = '''\
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The zephyr/module.yml file is a simple list of key value pairs to be used by
+# the build system.
+type: map
+mapping:
+  build:
+    required: true
+    type: map
+    mapping:
+      cmake:
+        required: false
+        type: str
+      kconfig:
+        required: false
+        type: str
+'''
+
+AUTO_GENERATED_HEADER = '# NOTE: THIS FILE IS AUTOGENERATED - DO NOT EDIT\n\n'
```

Comment:
Can we drop this?

Then we can drop the regex:

https://github.com/zephyrproject-rtos/zephyr/pull/14667/files#diff-af3b638bc2a3e6c650974192a53c7291R581

Users really should know that they can't modify any files in the build directory ...

Reply:
Fine with me, but it was originally introduced due to this comment: https://github.com/zephyrproject-rtos/zephyr/pull/13067#discussion_r254043635
> - the filename should reflect that, e.g. modules.generated.kconfig
> - Add a header of "# NOTE: THIS IS AUTOGENERATED BY " to this file.

Reply:
I disagree with that comment.

Most files in the build directory are generated (as opposed to copied from source, which is rarely done in CMake-based build systems). And most files do not have the word generated in the filename or in the header.

But I can understand having the disclaimer in generated C code for users with editors that don't make it obvious that they have opened a file in the build directory.

Reply:
> I disagree with that comment.

Personally this is not important to me, and per default I would not have put it in first place.
I was just referring to why it was there. 

So it's removed again.